lib/geminabox/gem_list_merge.rb

```diff
@@ -1,23 +1,38 @@
+require "set"
+
 module Geminabox
   class GemListMerge
     attr_accessor :list
 
     IGNORE_DEPENDENCIES = 0..-2
 
-    def self.from(*lists)
-      lists.map{|list| new(list)}.inject(:merge)
+    def self.merge(strategy, lists)
+      lists.map{|list| new(list)}.inject{|a,b| a.merge(b, strategy) }
     end
 
     def initialize(list)
       @list = list
     end
 
-    def merge(other)
+    def merge(other, strategy = nil)
+      send(strategy || :ours, other)
```

Comment:
Using `send` here introduces a new vulnerability as anything passed to `Geminabox.rubygems_proxy_merge_strategy` could be activated. With so few options there are better ways of doing this. For example:
```ruby
def merge(other)
  case strategy
  when :both
    both(other)
  when :theirs
    theirs(other)
  else
    ours(other)
  end
end
```

Reply:
Note that `Geminabox.rubygems_proxy_merge_strategy` is not untrusted input. Would put the concern about introducing a new vuln to rest.

The motivation here is to demonstrate plugging in a named merge strategy. Can implement what matches your sense of style for this codebase.